build_runner_core/lib/src/package_graph/package_graph.dart

```diff
@@ -72,13 +72,17 @@ class PackageGraph {
           'This program must be ran from the root directory of your package.');
     }
     final rootPubspec = _pubspecForPath(packagePath);
-    final rootPackageName = rootPubspec['name'] as String;
+    final rootPackageName = rootPubspec['name'] as String?;
+    if (rootPackageName == null) {
+      throw StateError('The current package has no name, please add one to the '
```

Comment:
would it be reasonable for us to call it `$root` or something along those lines and handle this case silently?

Reply:
I double checked and you're correct, `dart pub get` refuses to run without a package name. Throwing a StateError SGTM.